lib/errors/oauth-error.js

```diff
@@ -3,39 +3,40 @@
 /**
  * Module dependencies.
  */
-const util = require('util');
 const http = require('http');
 /**
  * Constructor.
  */
 
-function OAuthError(messageOrError, properties) {
-  let message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
-  const error = messageOrError instanceof Error ? messageOrError : null;
-  if (properties == null || !Object.entries(properties).length ) {
-    properties = {};
-  }
+class OAuthError extends Error {
+  constructor(messageOrError, properties) {
+    let message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
+    const error = messageOrError instanceof Error ? messageOrError : null;
+    if (properties == null || !Object.entries(properties).length) {
+      properties = {};
+    }
 
-  properties = Object.assign({ code: 500 }, properties);
+    properties = { code: 500, ...properties };
 
-  if (error) {
-    properties.inner = error;
-  }
-  if (!message || message.length === 0) {
-    message = http.STATUS_CODES[properties.code];
-  }
-  this.code = this.status = this.statusCode = properties.code;
-  this.message = message;
-  for (const key in properties) {
-    if (key !== 'code') {
-      this[key] = properties[key];
+    if (error) {
+      properties.inner = error;
+    }
+    if (!message || message.length === 0) {
+      message = http.STATUS_CODES[properties.code];
+    }
+
+    super(message, properties);
```

Comment:
Why did you place it exactly at this line and not after the for loop?

Reply:
Good question, I changed the place again, put it in the first line of the constructor